pym/gentoolkit/eclean/search.py

```diff
@@ -332,6 +332,18 @@ def _non_destructive(
             deprecated.update(_deprecated)
         return pkgs, deprecated
 
+    def git_check(self, distdir):
+        """Checks $DISTDIR/git3-src for checkouts which are not in the vardb"""
+        expected_dirs = [
+            self.vardb._aux_env_search(i, ["EGIT_DIR"])["EGIT_DIR"]
```

Comment:
EGIT_DIR contains an absolute path that starts with DISTDIR as it was defined at the time the package was installed.

It is possible for the user to move DISTDIR to another location by copying/moving the files and updating make.conf. In this case, eclean-dist would end up removing all git repos since the paths would not match in the new DISTDIR.

A simple workaround might be to take the last component of EGIT_DIR, and append it to the current value of `${DISTDIR}/git3-src`.

Reply:
Yes you are right @floppym ! I'll change that